### ISSUE TITLE
C API `var_atom_t` is confusing and unneeded

### ISSUE THREAD

**Issue by luketpeterson** [trueagi-io/hyperon-experimental#347]
This is a very minor point, but I think C API functions would be more orthogonal is they dealt individually with Variable Atoms and other bound Atoms, rather than bundling both together into a `var_atom_t`

This was probably my biggest point of confusion when I was first learning the API.  My assumption was that the `var_atom_t` was a single variable atom.  However it's actually a parallel way to express variable bindings.

This is not a big deal, but I feel like it would simplify the C API to unpack binding as variable-atom pairs.

If a single struct still makes sense to you, I propose the name `var_binding_t` instead.

**Comment by luketpeterson** [comment]
Thinking more about this - I actually don't mind that there is a C struct for a single Bindings.  I don't love it, but it doesn't bother me.

However, I am very bothered by the fact that people will think `var_atom_t` is a "variable atom type", which it absolutely is not.

**Comment by vsbogd** [comment]
`var_atom_t` is an attempt to represent one variable binding, and I agree the name is confusing. I also not sure it is a best way to represent bindings at the moment. As we now have `bindings_t` in C API may be we can replace `var_atom_t` by calling `Bindings::resolve()` method.

**Comment by luketpeterson** [comment]
I ended up leaving the API functional semantics alone, for now, around adding and traversing bindings.  I just changed the function signatures to accept and provide single var-atom pairs, to line up with Rust.

I think we will want to reevaluate Bindings and BindingsSet representation and traversal when we implement comprehensions and non-determinism / inference control.  But that's a big project and I don't have a totally clear vision yet.
